corehq/apps/api/es.py

```diff
@@ -676,6 +676,13 @@ def consume_params(self, raw_params):
     TermParam('xmlns', 'xmlns.exact'),
     TermParam('case_name', 'name', analyzed=True),
     TermParam('case_type', 'type', analyzed=True),
+    # terms listed here to prevent conversion of their values to lower case since
+    # since they are indexed as `not_analyzed` in ES
+    TermParam('type.exact'),
+    TermParam('name.exact'),
+    TermParam('external_id.exact'),
+    TermParam('contact_phone_number'),
```

Comment:
how does type, external_id and contact_phone_number be different if they are changed to lower case? aren't they already lower cased.

Reply:
If you use a term filter on an `analyzed` field then you must convert the filter value to lower case otherwise it won't match. 

If you use a term filter on an un-analyzed field then you must not change the case since you are searching for exact matches only.

All these fields are un-analyzed so we should not convert them to lower case.

Reply:
here's an example, let's say the case type is `Person` with upper case

if they search for `type=Person` and we don't convert it to lower case it won't match any records

but if they search for `type.exact=Person` it will match

and the inverse is also true, `type=person` matches but `type.exact=person` does not

Reply:
I think i get three of those four cases. Why does the nonexact `type=Person` not match

Reply:
It's because our default analyzers have use the `lowercase` filter: https://github.com/dimagi/commcare-hq/blob/3851b7a79b3883edfa54bdc8aa4254a7688f782f/corehq/ex-submodules/pillowtop/es_utils.py#L42